src/nat/middleware/dynamic/dynamic_function_middleware.py

```
@@ -0,0 +1,1069 @@
+            if callable(value) or inspect.iscoroutine(value):
+                return None
+            return value
+        except Exception:
+            return None
+
+    # ==================== Policy Orchestration ====================
+
+    async def function_middleware_invoke(self,
+                                         *args: Any,
+                                         call_next: CallNext,
+                                         context: FunctionMiddlewareContext,
```

Ruff (0.14.7)

Undefined name `FunctionMiddlewareContext`

(F821)